components/homme/src/share/control_mod.F90

```diff
@@ -230,4 +246,365 @@ module control_mod
+                write(iulog,*) 'dt_remap_factor and dt_tracer_factor were specified, &
+                     &but neither divides the other.'
+             end if
+             if (abort_in) call abortmp('timestep_make_parameters_consistent: divisibility error')
+             return
+          end if
+       end if
+       qsplit_prev = qsplit
+       rsplit_prev = rsplit
+       qsplit = dt_tracer_factor
+       ! This is the only inconsistent setting. But I want to keep
+       ! this here b/c almost all uses of rsplit is simply for whether
+       ! it's == 0, and I don't want to touch all those lines in this
+       ! PR.
+       if (dt_tracer_factor <= dt_remap_factor) then
+          rsplit = dt_remap_factor/dt_tracer_factor
+       else
+          ! If rsplit cannot be set consistently (because
+          ! dt_tracer_factor < dt_remap_factor), then just preserve
+          ! the sign to distinguish between vertically Eulerian and
+          ! Lagrangian methods.
+          if (dt_remap_factor > 0) then
+             rsplit = 1
+          else
+             rsplit = 0
+          end if
+       end if
+       if (split_specified .and. (qsplit /= qsplit_prev .or. rsplit /= rsplit_prev) .and. &
+            par%masterproc .and. .not. silent_in) then
+          write(iulog,'(a,i2,a,i2,a,i2,a,i2,a)') &
+               'dt_remap_factor and dt_tracer_factor were specified, changing qsplit from ', &
+               qsplit_prev, ' to ', qsplit, ' and rsplit from ', rsplit_prev, ' to ', rsplit, '.'
+       end if
+    end if
+
+    status = 0 ! success value
+  end function timestep_make_subcycle_parameters_consistent
+
+  function timestep_make_eam_parameters_consistent(par, dt_remap_factor, dt_tracer_factor, &
+       nsplit, nstep_factor, tstep, dtime, abort, silent) result(status)
+
+    use parallel_mod, only: abortmp, parallel_t
+    use kinds, only: iulog
+
+    type (parallel_t), intent(in) :: par
+    integer, intent(in) :: dt_remap_factor, dt_tracer_factor
+    integer, intent(inout) :: nsplit
+    integer, intent(out) :: nstep_factor
+    real(kind=real_kind), intent(inout) :: tstep
+    integer, intent(inout) :: dtime
+    logical, intent(in), optional :: abort, silent
+    integer :: status
+
+    real(kind=real_kind), parameter :: &
+         zero = 0.0_real_kind, &
+         eps = epsilon(1.0_real_kind), &
+         divisible_tol = 1e3_real_kind*eps
+
+    real(kind=real_kind) :: nsplit_real, tmp
+    integer :: dt_max_factor
+    logical :: abort_in, silent_in
+
+    abort_in = .true.
+    if (present(abort)) abort_in = abort
+    silent_in = .false.
+    if (present(silent)) silent_in = silent
+
+    status = -1 ! error value for early returns on error
+
+    !! Process dtime, tstep, nsplit.
+    dt_max_factor = max(dt_remap_factor, dt_tracer_factor)
+
+    ! Every 'if' has an 'else', so every case is covered.
+    if (nsplit > 0) nstep_factor = dt_max_factor*nsplit
+    if (dtime > 0) then
```

Comment:
I could not trace this: why would dtime be < 0?

Reply:
It is meaningless in standalone HOMME. For testing and possible future use, I want dtime=-1 to signify it should be deduced (e.g., if se_nsplit and time are specified, and dtime=-1).